ehr/src/org/labkey/ehr/table/DefaultEHRCustomizer.java

```diff
@@ -356,45 +228,152 @@ private void doSharedCustomization(AbstractTableInfo ti)
                     UserSchema us = getEHRStudyUserSchema(ti);
                     if (us != null)
                     {
-                        col.setFk(new QueryForeignKey(us, null, "Animal", "Id", "Id"));
+                        col.setFk(new QueryForeignKey(ti.getUserSchema(), ti.getContainerFilter(), us, null, "Animal", "Id", "Id"));
```

Comment:
would it be better to store ti.getUserSchema() and ti.getContainerFilter() in temp variables outside this for loop?

Reply:
Simple getters like this are super-fast, so it's not a perf concern. I'd consider this equally good code, so inclined to just keep it as-is unless I'm missing something.

Reply:
got it, thanks.